### PR TITLE
Standardise docs to recommended scheme

### PR DESCRIPTION
**Pull request by xross**
- Standardise documentation structure
- Standardise readme
- Remove superfluous readmes (duplicated text)
- Standardise some doc content
- Make doc headings uniform
- Moved most of the readme text into the main docs (wasn't suitable for website/IP browser landing page)
- Put test information at the end of the docs (users don't care about this, generally)
- Updated instructions for running tests to use xcommon-cmake
- (my editor automatically removed a load of white-space so you might want to review with "hide white-space" checked)

### CODE REVIEW

File: doc/rst/src/getting_started.rst [hunk 1]
```diff
@@ -1,71 +1,64 @@
...
---------
-
-This library makes use of the CMake build system. See the 
-`GitHub readme <https://github.com/xmos/lib_xcore_math>`_ for instructions on
-obtaining this library's source code and including it in your application's build.
+    #include "xmath/xmath.h"
 
 Usage
------
+======
```
Comment (danielpieczko) [inline doc/rst/src/getting_started.rst:56]: (nitpick) Underline is longer than necessary

File: doc/rst/src/reference/filter/filter_index.rst [hunk 2]
```diff
@@ -1,11 +1,11 @@
 .. _filter_api:
 
 Filtering API
-=============
+--------------
```
Comment (danielpieczko) [inline doc/rst/src/reference/filter/filter_index.rst:4]: Another long underline.

File: doc/rst/src/tests.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    xsim bin/bfp_tests.xe
+
+.. warning::
+
+    Running the unit tests in the simulator may be *very* slow.
+
+To execute the BFP unit tests built for an x86 host platform, configure the build using the
+``NATIVE_BUILD`` option: ::
+
+    cmake -B build_x86 -G "Unix Makefiles" -D BUILD_NATIVE=TRUE'
```
Comment (danielpieczko) [inline doc/rst/src/tests.rst:53]: Unwanted apostrophe at the end of this command

File: doc/rst/src/tests.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+**********
+Unit tests
+**********
+
+This project uses `XCommon CMake` to build the unit tests in a similar fashion to the examples.
+
+Unit tests target the `XK-EVK-XU316` board and x86 platforms.
+All unit tests are located in the */tests/* directory:
+
+* `/tests/ <https://github.com/xmos/lib_xcore_math/tree/develop/tests/>`_ - Unit test projects for ``lib_xcore_math``:
```
Comment (danielpieczko) [inline doc/rst/src/tests.rst:11]: I don't like the links to develop, as existing documents break if this is changes in the future. But that's just my opinion, and I can't suggest an alternative link, so this is fine as it is.

**Review by danielpieczko — APPROVED** [review]

File: doc/rst/src/getting_started.rst (hunk 1 above)
Comment (xross) [reply doc/rst/src/getting_started.rst:56]: nitpicking fine!! :)

File: doc/rst/src/tests.rst (hunk 4 above)
Comment (xross) [reply doc/rst/src/tests.rst:11]: Agreed. I actually removed all links to GitHub in the library description. I left these ones as they are a bit more "developery"

**Pull request merged**